genesyscloud/routing_settings/resource_genesyscloud_routing_settings.go

```diff
@@ -0,0 +1,216 @@
+
+	_ = d.Set("contactcenter", []interface{}{contactSettings})
+	return nil
+}
+
+func updateContactCenter(ctx context.Context, d *schema.ResourceData, proxy *routingSettingsProxy) diag.Diagnostics {
+	var removeSkillsFromBlindTransfer bool
+
+	if contactCenterConfig := d.Get("contactcenter"); contactCenterConfig != nil {
+		if contactCenterList := contactCenterConfig.([]interface{}); len(contactCenterList) > 0 {
+			contactCenterMap := contactCenterList[0].(map[string]interface{})
+
+			if contactCenterMap["remove_skills_from_blind_transfer"] != nil {
+				removeSkillsFromBlindTransfer = contactCenterMap["remove_skills_from_blind_transfer"].(bool)
+			}
+
+			contactCenterSettings := platformclientv2.Contactcentersettings{
+				RemoveSkillsFromBlindTransfer: &removeSkillsFromBlindTransfer,
+			}
+
+			resp, err := proxy.updateRoutingSettingsContactCenter(ctx, contactCenterSettings)
+			if err != nil {
+				return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update contact center for routing settings %s error: %s", d.Id(), err), resp)
+			}
+		}
+	}
+	return nil
+}
+
+func readRoutingSettingsTranscription(ctx context.Context, d *schema.ResourceData, proxy *routingSettingsProxy) diag.Diagnostics {
+	transcription, resp, getErr := proxy.getRoutingSettingsTranscription(ctx)
+	if getErr != nil {
+		if util.IsStatus404(resp) {
+			return nil
+		}
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to read contact center for routing settings %s error: %s", d.Id(), getErr), resp)
+	}
+
+	if transcription == nil {
+		_ = d.Set("transcription", nil)
+		return nil
+	}
+
+	transcriptionSettings := make(map[string]interface{})
+	resourcedata.SetMapValueIfNotNil(transcriptionSettings, "transcription", transcription.Transcription)
+	resourcedata.SetMapValueIfNotNil(transcriptionSettings, "transcription_confidence_threshold", transcription.TranscriptionConfidenceThreshold)
+	resourcedata.SetMapValueIfNotNil(transcriptionSettings, "low_latency_transcription_enabled", transcription.LowLatencyTranscriptionEnabled)
+	resourcedata.SetMapValueIfNotNil(transcriptionSettings, "content_search_enabled", transcription.ContentSearchEnabled)
+	resourcedata.SetMapValueIfNotNil(transcriptionSettings, "pci_dss_redaction_enabled", transcription.PciDssRedactionEnabled)
+	resourcedata.SetMapValueIfNotNil(transcriptionSettings, "pii_redaction_enabled", transcription.PiiRedactionEnabled)
+
+	_ = d.Set("transcription", []interface{}{transcriptionSettings})
+	return nil
+}
+
+func updateTranscription(ctx context.Context, d *schema.ResourceData, proxy *routingSettingsProxy) diag.Diagnostics {
+	transcriptionRequest := platformclientv2.Transcriptionsettings{}
+
+	if transcriptionConfig := d.Get("transcription"); transcriptionConfig != nil {
+		if transcriptionList := transcriptionConfig.([]interface{}); len(transcriptionList) > 0 {
+			transcriptionMap := transcriptionList[0].(map[string]interface{})
```

Comment:
You could reduce nesting here by doing 

```golang
if transcriptionConfigList, ok := d.Get("transcription").([]interface{}); ok && len(transcriptionConfigList) > 0 {
```

The nested if isn't necessary

Reply:
Also there are no required attributes in the schema for `transcription` , so we should be checking that the item isn't nil 

```golang
transcriptionMap, ok := transcriptionList[0].(map[string]interface{})
if !ok {
  return nil
}
```